Fixes compilation and some tests

rshrotey-IBM opened the pull request:
Verified compilation
libarchive_test: 614/650 tests run

Knows issues:
1. Not all tests run at this point, its a WIP
2. Running zopen build -f -vv results in an error in the configure stage It can be bypassed by rerunning zopen build -vv. This will be fixed in future.

rshrotey-IBM committed:
Fixes compilation and some tests

Verified compilation
libarchive_test: 614/650 tests run

Knows issues:
1. Not all tests run at this point, its a WIP
2. Running zopen build -f -vv results in an error in the configure stage
   It can be bypassed by rerunning zopen build -vv. This will be fixed
   in future.

Signed-off-by: Rahul Shrotey <Rahul.Shrotey@ibm.com>

IgorTodorovskiIBM commented on buildenv:
```diff
@@ -0,0 +1,44 @@
+export ZOPEN_DEV_URL="https://github.com/libarchive/libarchive.git"
```
Can you add the bump line for automatic updating? E.g https://github.com/zopencommunity/gitport/blob/main/buildenv#L5-L12

rshrotey-IBM replied:
Done

IgorTodorovskiIBM commented on buildenv:
```diff
@@ -0,0 +1,44 @@
+export ZOPEN_DEV_URL="https://github.com/libarchive/libarchive.git"
+export ZOPEN_DEV_DEPS="zoslib git curl make autoconf tar lz4 cpio openssl zstd automake libtool"
+export ZOPEN_BUILD_LINE="DEV"
```
We usually aim for STABLE as the default build line unless there is a problem with the stable line.
You can also specify a stable release with export ZOPEN_STABLE_TAG="v3.7.9"

rshrotey-IBM replied:
Done

IgorTodorovskiIBM commented on buildenv:
```diff
@@ -0,0 +1,44 @@
+export ZOPEN_COMP=clang
+export ZOPEN_BOOTSTRAP="./build/autogen.sh"
+export ZOPEN_EXTRA_CFLAGS="${ZOPEN_EXTRA_CFLAGS} -mzos-target=zosv2r5 -D_POSIX_C_SOURCE=200809L -D_XPLATFORM_SOURCE=1"
+export ZOPEN_EXTRA_CPPFLAGS="${ZOPEN_EXTRA_CPPFLAGS}  -mzos-target=zosv2r5"
```
Since we're targetting v2r5, can you add a prereq line like this: https://github.com/zopencommunity/redisport/blob/main/buildenv#L14

rshrotey-IBM replied:
Done

IgorTodorovskiIBM commented on patches/libarchive/test/test_read_disk_directory_traversals.c.patch:
```diff
@@ -0,0 +1,37 @@
+ 	/* Save current working directory. */
+ #if defined(PATH_MAX) && !defined(__GLIBC__)
++#ifdef __MVS__
++	initial_cwd = getcwd(buf, PATH_MAX);
```
Was this due to a compile issue or runtime issue?

rshrotey-IBM replied:
It was a runtime issue. getcwd returned NULL leading to segmentation fault.

IgorTodorovskiIBM commented on buildenv:
```diff
@@ -0,0 +1,44 @@
+  echo "actualFailures:0"
+  echo "totalTests:1"
+  echo "expectedFailures:0"
+  echo "expectedTotalTests:1"
```
I noticed that you have test case patches. Do you plan to add logic here to parse out the test results?

rshrotey-IBM replied:
Yes, I've added the logic. Change is in the next commit. This is the output.
```
VERBOSE: Test status:
totalTests:475
actualFailures:107
skippedTests:232
expectedFailures:0
VERBOSE: Test results: 368 tests pass out of 475 tests - 77.47% success rate
```

IgorTodorovskiIBM commented on buildenv:
```diff
@@ -0,0 +1,44 @@
+{
+  # Modify to echo the version of your tool/library
+  # Rather than hardcoding the version, obtain the version by running the tool/library
+  echo "1.0.0"
```
You can update this to the stable version if you add the ZOPEN_STABLE_TAG

rshrotey-IBM replied:
Done.

rshrotey-IBM committed:
* Incorporate review comments

Signed-off-by: Rahul-Shrotey <Rahul.Shrotey@ibm.com>

rshrotey-IBM committed:
* Fix bsdcpio_test and ./manual_tests.sh
* Update README
* Current test status
* Remove unwanted patches

VERBOSE: Test status:
totalTests:501
actualFailures:169
skippedTests:236
expectedFailures:0
VERBOSE: Test results: 332 tests pass out of 501 tests - 66.27% success rate

Signed-off-by: Rahul-Shrotey <Rahul.Shrotey@ibm.com>

rshrotey-IBM committed:
* Add manual_tests.sh script

Signed-off-by: Rahul-Shrotey <Rahul.Shrotey@ibm.com>

IgorTodorovskiIBM reviewed (approved):
LGTM, thank you!

rshrotey-IBM commented:
Thanks Igor. I don't have write access, so I won't be able to merge the PR.
